src/webview/memory-webview-view.tsx

```diff
@@ -135,21 +153,52 @@ class App extends React.Component<{}, MemoryAppState> {
         messenger.onRequest(getWebviewSelectionType, () => this.getWebviewSelection());
         messenger.onNotification(showAdvancedOptionsType, () => this.showAdvancedOptions());
         messenger.sendNotification(readyType, HOST_EXTENSION, undefined);
+        this.updatePeriodicRefresh();
     }
 
-    public componentDidUpdate(_: {}, prevState: MemoryAppState): void {
-        const addressPaddingNeedsUpdate =
-            (this.state.addressPadding === 'Min' && this.state.memory !== prevState.memory)
-            || this.state.addressPadding !== prevState.addressPadding;
-        if (addressPaddingNeedsUpdate) {
+    public componentDidUpdate(_: {}, from: MemoryAppState): void {
+        const current = this.state;
+        const stateChange: Change<MemoryAppState> = { from, to: current };
+        const sessionContextChange: Change<SessionContext> = { from: from.sessionContext, to: current.sessionContext };
+
+        if (hasChanged(stateChange, 'addressPadding') || (this.state.addressPadding === 'Minimal' && hasChanged(stateChange, 'memory'))) {
             const effectiveAddressLength = this.getEffectiveAddressLength(this.state.memory);
             if (this.state.effectiveAddressLength !== effectiveAddressLength) {
                 this.setState({ effectiveAddressLength });
             }
         }
+        if (hasChanged(stateChange, 'periodicRefresh') || hasChanged(stateChange, 'periodicRefreshInterval') || hasChanged(sessionContextChange, 'stopped')) {
+            this.updatePeriodicRefresh();
+        }
+
+        if (current.refreshOnStop === 'on' && hasChangedTo(sessionContextChange, 'stopped', true)) {
+            this.fetchMemory();
+        }
+        // activate the code below if you want to refresh the memory when the view becomes active (focussed)
+        // const viewStateChange: Change<ViewState> = { from: from.viewState, to: current.viewState };
+        // if (hasChangedTo(viewStateChange, 'active', true)) {
+        //     this.fetchMemory();
+        // }
```

Comment:
It looks like this is where the view state would be accessed. Given that we've decided not to include this as an option available to user, do we want to include this code?

Reply:
Initially I wanted to keep it until we all agreed on the options we want to have. As we did that now, I removed all view state-related code to not pollute our code base, thank you for catching that!